subsys/logging/log_backend_uart.c

```diff
@@ -103,14 +104,15 @@ static void process(const struct log_backend *const backend,
 {
 	uint32_t flags = log_backend_std_get_flags();
 
-	flags |= IS_ENABLED(CONFIG_LOG_BACKEND_UART_SYST_ENABLE) ? LOG_OUTPUT_FLAG_FORMAT_SYST : 0;
+	void (*log_output_func)(const struct log_output*, struct log_msg2*,
```

Comment:
you probably wanted to add it to log_msg2 process function, this one is deprecated. When you extend log_output api with getter function you can use it here.

Reply:
@nordic-krch  Did not get it. What is deprecated? If the function pointer table is called from log_output_msg2_process() function, then there would be no use of the table since the table selects the routine (log_output_msg2_process/ log_output_msg2_syst_process etc) that needs to be called based on Kconfig value for formats. Not using "flags" for setting of log format type as per this implementation.
Also, when you say log_output api, do you refer to log_output structure or log_output.c?

Reply:
sorry, i got confused and did not notice the gap in the file so i thought that changes are in `put` function.

Reply:
No problem. Thanks for your detailed review though. I am working on addressing the other suggested changes.